### ISSUE TITLE
[bitnami/harbor] - potential leak of password for External Redis

### ISSUE THREAD

**Issue by jowko** [bitnami/charts#7677]
**Which chart**:
bitnami/harbor 11.0.3

**Describe the bug**
External Redis password can leak easily.

**To Reproduce**

Run Harbor Helm Chart with default settings. I only modified Redis settings to point to external Redis instance.

**Expected behavior**
Redis password should be put in k8s Secret.
Most deployments have this password in secrets.
But for `harbor-jobservice` this password is placed in ConfigMap: `harbor-jobservice`.

The second issue is that password to Redis is printed in some logs (in some cases entries are masked).
Examples:
```
[DEBUG] [/lib/redis/redisclient.go:66]: get redis pool:cefffaa679bac19aa1630e21abcd18e8redis://redis:<MY_PASSWORD>@<MY_HOSTNAME>:<MY_PORT>/0
[ERROR] [/lib/cache/cache.go:110]: failed to ping redis://redis:xxxxx@<MY_HOSTNAME>:<MY_PORT>/0, retry after 500ms : dial tcp: lookup <MY_HOSTNAME> on <MY_IP>: no such host
```

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.6.3", GitCommit:"d506314abfb5d21419df8c7e7e68012379db2354", GitTreeState:"clean", GoVersion:"go1.16.5"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.4", GitCommit:"3cce4a82b44f032d0cd1a1790e6d2f5a55d20aae", GitTreeState:"clean", BuildDate:"2021-08-11T18:16:05Z", GoVersion:"go1.16.7", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.5+IKS", GitCommit:"3ad2d38ee2ef808e6700d68e7feb1a4a31f8c3d0", GitTreeState:"clean", BuildDate:"2021-09-17T06:19:10Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
I run Harbor on IBM Cloud Kubernetes. External Redis is managed by IBM Cloud too.

**Comment by Mauraza** [comment]
Hi @jowko,

About Redis who did you pass the password? 
And for `harbor-jobservice` exist https://github.com/bitnami/charts/blob/master/bitnami/harbor/templates/jobservice/jobservice-secrets.yaml your `harbor-jobservice` is not using this secret?

**Comment by jowko** [comment]
Hi @Mauraza 
I'm not sure if I understand your question, but I did not pass Redis password to anyone.
I just observed potential password leak:
- All k8s users which can read ConfigMap can read Redis password. `harbor-jobservice` uses both secrets and config maps. The leak is located in https://github.com/bitnami/charts/blob/master/bitnami/harbor/templates/jobservice/jobservice-cm.yaml file.
There is `config.yml` which contains entry `worker_pool.redis_pool.redis_url`. This line has such format:
```
redis_url: \"redis://redis:<PASSWORD>@<HOSTNAME>:<PORT>/1\"
```
- All people with access to our Harbor logs can also see Redis password, since it is logged as plain text. Some of these logs statements are obfuscating password from connection url (see second line from logs example in first post). We should check the code related to Redis and obfuscate password in every case when password or redis url is logged.
